### DISCUSSION TITLE
Shaping points disappear after route calculation

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#410] [Q&A] [answered]
Don't know exactly when it happens, but sometimes shaping points disappear after route calculation. If you then close the app and open it again, they appear again.

**Comment by devemux86** [comment]
@DolomitesTom thanks for the report.

Has the previous route calculation finished or you're making changes while the previous calculation is in progress?
Did it happen with shaping points or all intermediate points (via + shaping)?

What route change are you making when this happens? Move waypoints, rename, ...?

**Reply by DolomitesTom** [reply]
Good questions, have to pay attention to. I was adapting a loaded route, deleting and moving SPs after changing all WPs into SPs and suddenly the SPs are not visible any more, also when zooming in.

**Comment by DolomitesTom** [comment]
It happened again. This time after changing waypoints into shaping points via menu "Change all" in the waypoint list, all waypoints disappeared and after closing and opening Cruiser again, they are all there again.
<img src="https://github.com/devemux86/cruiser/assets/121371629/24ec5ad1-5f15-415e-8aa8-016511485ce7" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/121371629/a6295c9f-f035-4515-b072-a5ae70f4ef1e" width="50%"/>

**Comment by devemux86** [comment]
@DolomitesTom were the start + end points visible on the map and only the shaping points were missing?
Or were all waypoints missing (start + shaping + end)?

**Reply by DolomitesTom** [reply]
Start and end point were visible, only all SPs diappeared.i could just reproduce it.

**Reply by devemux86** [reply]
@DolomitesTom with what process, the waypoints list?

I cannot reproduce it. Can you send me your profile and the route?

**Reply by DolomitesTom** [reply]
yes, waypoint list and then change all on one of the SPs.

Can you please send me a private email which I can answer? Don't find your email address.

It happens when I load a tour from my locations -- routes (example route has  waypoints) and then move SPs to force recalculation or if I add and delete SPs. But not always, it happens suddenly. That makes it diffucult to reproduce, but it happens.

**Reply by devemux86** [reply]
@DolomitesTom you can use the email you have sent me before, it is on [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser) at "App support".

**Comment by devemux86** [comment]
Next Cruiser will have improvements to some route planning actions:
- Change name of waypoints
- Change type of waypoints
- Split route

@DolomitesTom we will see how it improves this case.

**Reply by DolomitesTom** [reply]
Till now I didn't recognize this effect anymore with 38.b1

**Reply by devemux86** [reply]
@DolomitesTom please try Cruiser 3.0.38-**b2** with the mentioned improvements:
- https://github.com/devemux86/cruiser/discussions/414

**Reply by DolomitesTom** [reply]
Till now also no problems so far with version 38.b2. With 38 I could simulate it pretty quick.

**Reply by devemux86** [reply]
@DolomitesTom the improvement is in 38-b2.
(38-b1 was the same as release 37 in this area)

**Comment by devemux86** [comment, marked as answer]
Fixed in [Cruiser 3.0.38](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by DolomitesTom** [reply]
Have again this issue that all shaping and waypoints disappear after calculation in 3.0.40.

**Reply by devemux86** [reply]
@DolomitesTom thanks for the report.

Please provide more details about exactly what you were doing when this happened.

Also the specific route.

**Reply by devemux86** [reply]
@DolomitesTom

Did you perform route calculations while the previous calculation was not complete?
Or did you zoom in/out the map at the same time?

The next update will contain an improvement, so let's see if it happens again.

**Comment by DolomitesTom** [comment]
Yes both is true, since the calculation took that long, 750 km tour.
Yes, can confirm that I also zoomed in and out.

**Reply by devemux86** [reply]
@DolomitesTom thanks, this is very useful information. :+1:

**Reply by devemux86** [reply]
@DolomitesTom Forgot to mention, please try the latest Beta and see if it still happens:
- https://github.com/devemux86/cruiser/discussions/451

**Reply by DolomitesTom** [reply]
Till now not seen this any more
